src/test/java/com/artipie/rpm/pkg/PrimaryOutputTest.java

```diff
@@ -0,0 +1,115 @@
+
+import com.artipie.rpm.Digest;
+import com.jcabi.log.Logger;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.nio.channels.FileChannel;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.logging.Level;
+import org.hamcrest.MatcherAssert;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import org.redline_rpm.ReadableChannelWrapper;
+import org.redline_rpm.Scanner;
+import org.redline_rpm.header.Format;
+import org.redline_rpm.header.Header;
+import org.xmlunit.matchers.CompareMatcher;
+
+/**
+ * Test for {@link PrimaryOutput}.
+ * @since 0.10
+ * @todo #241:30min Extend this test: add test methods to check `file()`, `maid()` and `tag()`
+ *  methods of `PrimaryOutput` and add method to check generated primary.xml for libdeflt test
+ *  rpm package.
+ */
+@SuppressWarnings("PMD.AvoidDuplicateLiterals")
+class PrimaryOutputTest {
+
+    @Test
+    void createsPrimaryForAbc(@TempDir final Path temp) throws IOException {
+        final Path res = temp.resolve("primary.xml");
+        try (PackageOutput.FileOutput primary = new PrimaryOutput(res)) {
+            primary.start();
+            final Path rpm =
+                Paths.get("src/test/resources-binary/abc-1.01-26.git20200127.fc32.ppc64le.rpm");
+            primary.accept(new FilePackage.Headers(this.header(rpm), rpm, Digest.SHA256));
+        }
+        MatcherAssert.assertThat(
+            new String(Files.readAllBytes(res), StandardCharsets.UTF_8),
+            CompareMatcher.isIdenticalTo(
+                new String(
+                    Files.readAllBytes(
+                        Paths.get("src/test/resources-binary/repodata/abc-primary.xml")
+                    ),
+                    StandardCharsets.UTF_8
+                )
+            ).ignoreWhitespace()
+            .ignoreElementContentWhitespace()
+            .normalizeWhitespace()
+            .withNodeFilter(
+                node -> !"file".equals(node.getLocalName())
+                    && !"provides".equals(node.getLocalName())
+                    && !"requires".equals(node.getLocalName())
+            ).withAttributeFilter(
+                attr -> !"file".equals(attr.getName()) && !"archive".equals(attr.getName())
+            )
+        );
+    }
+
+    /**
+     * Get header.
+     * @return The header
+     * @param file Rpm file path
+     * @throws InvalidPackageException In case package is invalid.
+     * @throws IOException On error
+     * @todo #241:30min Introduce class from this method: the method is copied from FilePackage
+     *  class, we need to extract class from it, test it and use it at least in FilePackage,
+     *  ModifiableMetadataTest and here.
+     */
+    @SuppressWarnings("PMD.AvoidCatchingGenericException")
+    private Header header(final Path file) throws InvalidPackageException, IOException {
```

Comment:
@olenagerasimova `InvalidPackageException` doesn't need to be declared explicitly. Also I don't think that is a good idea to throw both runtime and catch exceptions. I would throw only `InvalidPackageException` or only `IOException`. wdyt?